FirebaseDynamicLinks/Sources/FIRDynamicLinks.m

```diff
@@ -427,9 +432,21 @@ - (nullable FIRDynamicLink *)dynamicLinkFromUniversalLinkURL:(NSURL *)url {
       }
     }
   }
+  if (completion) {
```

Comment:
I'm not sure whether this completion block calling is useful here since it would just invoke the block without giving any clue on which url we have tried and failed and so i believe this can be removed. thoughts @christibbs @paulb777 ?

Reply:
It may be needed to satisfy the API if it expects there to be a callback.

However, as this is written, there could be two callbacks if the nested if is reached above and that doesn't seem right.

Reply:
@paulb777 We return on line 430, which should prevent 422 and 436 from both being fired.

Reply:
I agree with you @paulb777; but i think the 3p users can't do anything with the callback as they don't know to the minimum what url we have tried to parse.

I could be wrong but I'm not seeing a chance for two callback method invocation since we are returning right after the service call in line 425.

Reply:
Oh right, agreed that there can't be two callbacks.  I missed that return.

On the other question, couldn't 3p users check for the nil parameters to know that the url failed to parse or that `url.query.length` was 0?